hazelcast/src/test/java/com/hazelcast/serialization/SerializationBenchmark.java

```diff
@@ -0,0 +1,166 @@
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.nio.serialization.Portable;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OperationsPerInvocation;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import java.io.Externalizable;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.concurrent.TimeUnit;
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@Fork(value = 1, warmups = 0)
+@Warmup(iterations = 3, time = 1)
+@Measurement(iterations = 3, time = 1)
+@OperationsPerInvocation(SerializationBenchmark.BATCH_SIZE)
+public class SerializationBenchmark {
+    static final int BATCH_SIZE = 1_000;
+
+    Serializable serializable;
+    Serializer serializableSerializer;
+
+    Externalizable externalizable;
+    Serializer externalizableSerializer;
+
+    DataSerializable dataSerializable;
+    Serializer dataSerializableSerializer;
+
+    IdentifiedDataSerializable identifiedDataSerializable;
+    Serializer identifiedDataSerializableSerializer;
+
+    Portable portable;
+    Serializer portableSerializer;
+
+    Object object;
+    Serializer streamSerializer;
+
+    @Setup
+    public void setup() {
+        serializable = SerializableObject.object();
+        serializableSerializer = new Serializer(SerializableObject.serializationService());
+
+        externalizable = ExternalizableObject.object();
+        externalizableSerializer = new Serializer(ExternalizableObject.serializationService());
+
+        dataSerializable = DataSerializableObject.object();
+        dataSerializableSerializer = new Serializer(DataSerializableObject.serializationService());
+
+        identifiedDataSerializable = IdentifiedDataSerializableObject.object();
+        identifiedDataSerializableSerializer = new Serializer(IdentifiedDataSerializableObject.serializationService());
+
+        portable = PortableObject.object();
+        portableSerializer = new Serializer(PortableObject.serializationService());
+
+        object = StreamObject.object();
+        streamSerializer = new Serializer(StreamObject.serializationService());
+    }
+
+    @Benchmark
+    public void serializable(Blackhole blackhole) throws Exception {
+        byte[] serialized = serializableSerializer.serialize(serializable);
+        blackhole.consume(serializableSerializer.deserialize(serialized));
```

Comment:
I believe that in such straightforward scenarios you can just return the result instead of using `Blackhole`

Reply:
Yes. It makes the code a bit simpler to digest.